Examples/Io/Root/src/RootSimHitReader.cpp

```diff
@@ -149,7 +148,17 @@ ProcessCode RootSimHitReader::read(const AlgorithmContext& context) {
 
     const Acts::GeometryIdentifier geoid{m_uint64Columns.at("geometry_id")};
     const SimBarcode pid =
-        SimBarcode().withData(*m_vecUint32Columns.at("barcode"));
+        SimBarcode()
```

Comment:
I'd still make this backwards compatible if not too hard, what do you think?

Reply:
I think what @ntadej wanted to do is writing these branches from uproot and then read them back into ACTS. I think it is generally not a bad idea to split the `std::vector` into named individual branches. But potentially we should only keep one version since redundancy may lead to inconsistency. What @ntadej wants to do would immediately result in inconsistency.

Of course, alternatively we could leave everything as is and one has to use C++ or the original Python ROOT interface to manipulate/generate the defined format.

Reply:
I think for writing it is fine to just split and drop the old branch. For reading I would appreciate if it was kept backwards compatible for a bit (or I can just make it myself in my fork).

Reply:
I think I removed the combined branches altogether now. If you want the reader to be backward compatible, at runtime, it'll have to check which branches are in the input file, and then use those, right?

Reply:
Yes, note that the current implementation of the reader gracefully handles missing branches if they are not used (just a runtime warning).

But here as we would want to "unpack" the vector of barcodes an explicit runtime check would be needed.

Reply:
Thanks! Will test and report back.